src/substrait/textplan/README.md

```diff
@@ -0,0 +1,208 @@
+may not be optional.
+
+### Integer Literals
+
+There are four different integer widths.  While other languages provide for binary, octal, and hexadecimal
+representations of their integer literals (as well as spacing separators) human readability of literals isn't a current
+requirement of the language and so are omitted for brevity.
+
+#### Example Integer Literals
+```
+123_i8
+0_i16
+123_i32
+1234_i64
+```
+
+### Floating Point Literals
+
+Floating point literals are represented with by a decimal number followed by an optional exponent.   
+
+#### Example Floating Point Literals
+
+```
+-1_fp32
+0.3_fp64
+1.99E-13_fp64
+```
+
+### Decimal Literals
+
+Decimal literals are similar to that of floating point literals however are represented by a real number with an
+exponent which is specified as part of the type.  The type includes the precision and the scale (number of base 10
+digits to translate the result by). 
+
+#### Example Decimal Literals
+
+```
+-123_decimal<3,0>
+42_decimal<5,-4>
+```
+
+### String Literals
+
+No need for multiline.  Raw strings may be useful for regexes.  Strings with no size limit may be specified without a
+type specifier for brevity.
+
+| Escape Sequence | Description                         |
+|-----------------|-------------------------------------|
+| \n              | linefeed (ASCII code 10)            |
+| \r              | carriage return (ASCII code 13)     |
+| \t              | tab (ASCII code 8)                  |
+| \\              | backslash                           |
+| \'              | single quote                        |
+| \"              | double quote                        |
+| \xNN            | two byte hexadecimal character code |
+| \u{NNNNNNN}     | unicode code sequence               |
+
+#### Example String Literals
+
+```
+"simple text"
+"two\nlines with \"escapes\""_varchar<80>
+"abcde"_fixedchar<5>
+"unicode chars: \xA9 \u{023B}"
+```
+
+### Binary Literals
+
+Binary literals are represented as strings containing hexadecimal digits (no escapes required).  Fixed binary literals
+are measured in bytes and thus have two hexadecimal characters per required byte.  A UUID is effectively a
+fixedbinary\<16> that ignores any provided dashes in its representation. 
+
+#### Example Binary Literals
+
+```
+"0123456789abcde"_binary
+"1234"_fixedbinary<2>
+"00-ABCD-1234"_uuid
+```
+
+### Time Literals
+
+There are two kinds of time literals:  literals that represent particular points in time and literals that represent
+durations.  The timestamps represent dates from the beginning of the year 1000 to the end of the year 9999.  The 
+interval_year duration has its precision in months and the interval_day duration has its precision in microseconds.
+
+To make the structured data easier to read the pseudo types ```year```, ```month```, ```day```, ```second```, and
+```microsecond``` (plural or singular) are suggested for readability.
```

Comment:
Suggested?  Or required?

Reply:
Similar to structured literals in the following section types aren't needed because the type identifier already explains what is inside.  Effectively an interval_year or interval_day is merely a well-known structure.  If one only wanted to provide part of the structure the types would again be necessary (although now that I'm thinking about it we're actually providing names for this purpose here).